core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder requestTimeoutMillis(long requestTimeoutMillis) {
+        return (AnnotatedServiceBindingBuilder) super.requestTimeoutMillis(requestTimeoutMillis);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder maxRequestLength(long maxRequestLength) {
+        return (AnnotatedServiceBindingBuilder) super.maxRequestLength(maxRequestLength);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder verboseResponses(boolean verboseResponses) {
+        return (AnnotatedServiceBindingBuilder) super.verboseResponses(verboseResponses);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder requestContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.requestContentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder responseContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.responseContentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreview(int length) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreview(length);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreview(int length, Charset defaultCharset) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreview(length, defaultCharset);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder accessLogFormat(String accessLogFormat) {
+        return (AnnotatedServiceBindingBuilder) super.accessLogFormat(accessLogFormat);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder accessLogWriter(AccessLogWriter accessLogWriter,
+                                                          boolean shutdownOnStop) {
+        return (AnnotatedServiceBindingBuilder) super.accessLogWriter(accessLogWriter, shutdownOnStop);
+    }
+
+    /**
+     * Registers the {@link ServiceConfigBuilder} that is created from this Builder and returns
+     * {@link ServerBuilder}.
+     * @param service annotated service object to handle incoming requests matching path prefix, which
+     *                  can be configured through {@link AnnotatedServiceBindingBuilder#pathPrefix(String)}.
+     *                  If path prefix is not set then this service is registered to handle requests matching
+     *                  {@code /}
+     * @return {@link ServerBuilder} to continue building {@link Server}
+     */
+    public ServerBuilder buildAnnotated(Object service) {
+        final ImmutableList<ExceptionHandlerFunction> exceptionHandlerFunctions =
```

Comment:
Could you change this `ImmutableList` to `List` and the following assignments?
```suggestion
        final List<ExceptionHandlerFunction> exceptionHandlerFunctions =
```